src/dxtbx/format/FormatNXmx.py

```diff
@@ -15,15 +15,14 @@ class FormatNXmx(FormatNexus):
     def understand(image_file):
         with h5py.File(image_file) as handle:
             return (
-                len(
+                bool(
                     [
                         entry
                         for entry in dxtbx.nexus.nxmx.find_class(handle, "NXentry")
                         if "definition" in entry
                         and dxtbx.nexus.nxmx.h5str(entry["definition"][()]) == "NXmx"
                     ]
                 )
```

Comment:
Something I should have mentioned on the first review — I can't be sure I know of all the uses of the `understand` method of format classes, but from what I've seen, it looks like it doesn't need to return a `bool`, so the `bool` call may be unnecessary here.  I added it in my previous suggestion to strictly preserve the `len(...) > 0` behaviour, but duck typing exists for a reason, so maybe we just want
```suggestion
                [
                    entry
                    for entry in dxtbx.nexus.nxmx.find_class(handle, "NXentry")
                    if "definition" in entry
                    and dxtbx.nexus.nxmx.h5str(entry["definition"][()]) == "NXmx"
                ]
```

Reply:
Although format classes pre-date typing, the [docstring](https://github.com/dials/dxtbx/blob/eb039540a860b1d735d7613c9183054838ea86cf/src/dxtbx/format/Format.py#L106-L124) implies that the return type should be `bool` so I would probably prefer to keep it explicit rather than implicit (as that would break any hypothetical future type annotations).

Reply:
OK, that's cool too.